common/djangoapps/third_party_auth/tests/test_models.py

```diff
@@ -17,39 +25,61 @@ def setUp(self):
         super().setUp()
         self.saml_provider_config = SAMLProviderConfigFactory()
 
-    def test_unique_entity_id_enforcement_for_non_current_configs(self):
-        """
-        Test that the unique entity ID enforcement does not apply to noncurrent configs
-        """
-        with self.assertLogs() as ctx:
-            assert len(SAMLProviderConfig.objects.all()) == 1
-            old_entity_id = self.saml_provider_config.entity_id
-            self.saml_provider_config.entity_id = f'{self.saml_provider_config.entity_id}-ayylmao'
-            self.saml_provider_config.save()
-
-            # check that we now have two records, one non-current
-            assert len(SAMLProviderConfig.objects.all()) == 2
-            assert len(SAMLProviderConfig.objects.current_set()) == 1
-
-            # Make sure we can use that old entity id
-            SAMLProviderConfigFactory(entity_id=old_entity_id)
-
-            # 7/21/22 : Disabling the exception on duplicate entity ID's because of existing data.
-            # with pytest.raises(IntegrityError):
-            bad_config = SAMLProviderConfig(entity_id=self.saml_provider_config.entity_id)
-            bad_config.save()
-        assert ctx.records[0].msg == f'Entity ID: {self.saml_provider_config.entity_id} already in use'
-
-    @override_settings(FEATURES={'ENABLE_UNICODE_USERNAME': False})
-    def test_clean_username_unicode_disabled(self):
-        """
-        Test the username cleaner function with unicode disabled
-        """
-        assert clean_username('ItJüstWòrks™') == 'ItJ_stW_rks'
+    @ddt.data(
+        ('ItJüstWòrks™', False, 'ItJ_stW_rks'),
+        ('ItJüstWòrks™', True, 'ItJüstWòrks'),
+        ('simple_username', False, 'simple_username'),
+        ('simple_username', True, 'simple_username'),
+        ('test@example.com', False, 'test_example_com'),
+        ('test@example.com', True, 'test@example.com'),
+    )
+    @ddt.unpack
+    def test_clean_username(self, input_username, unicode_enabled, expected_output):
+        """Test the username cleaner function with different unicode settings."""
+        with override_settings(FEATURES={'ENABLE_UNICODE_USERNAME': unicode_enabled}):
+            self.assertEqual(clean_username(input_username), expected_output)
+
 
-    @override_settings(FEATURES={'ENABLE_UNICODE_USERNAME': True})
-    def test_clean_username_unicode_enabled(self):
+@ddt.ddt
+class TestSAMLConfigurationSignals(TestCase):
+    """
+    Tests for SAML configuration signal handlers and their effect on provider configs.
+    """
+
+    def setUp(self):
         """
-        Test the username cleaner function with unicode enabled
+        Set up test data.
         """
-        assert clean_username('ItJüstWòrks™') == 'ItJüstWòrks'
+        self.site = Site.objects.get_current()
+        # Create initial SAML configuration
+        self.saml_config = SAMLConfiguration.objects.create(
```

Comment:
Again, should we be using the factories like in: https://github.com/openedx/edx-platform/blob/0b7cd8bc0923161a620bb648339ecbb35980eb95/common/djangoapps/third_party_auth/management/commands/tests/test_saml.py#L71-L72?